streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java

```diff
@@ -63,11 +62,11 @@ public RackAwareTaskAssignor(final Cluster fullMetadata,
                                  final AssignmentConfigs assignmentConfigs) {
         this.fullMetadata = fullMetadata;
         this.partitionsForTask = partitionsForTask;
-        this.racksForProcessConsumer = racksForProcessConsumer;
         this.internalTopicManager = internalTopicManager;
         this.assignmentConfigs = assignmentConfigs;
         this.racksForPartition = new HashMap<>();
         this.racksForProcess = new HashMap<>();
+        validateClientRack(racksForProcessConsumer);
```

Comment:
I think we should document that we might throw here, because when we create ` RackAwareTaskAssignor` we need to catch this exception to ensure to not kill the `StreamThread`?

Now that I am realizing this, I am actually wondering if it's a good idea to handle it this way? In the end, for the callee it might be better to figure out if the `RackAwareTaskAssignor` can we used or not via method call before we create it? Maybe we nee some `static` method? -- Any other idea to make it more natural for the callee?

Reply:
Yeah. Reverted and introduced `validClientRack` to track it